crates/sui-indexer-alt-jsonrpc/src/data/read_error.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use diesel::result::Error as DieselError;
+
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum ReadError {
```

Comment:
I thought about whether we wanted separate errors for pg/bigtable but went with one unified error type for simplicity. And I moved this out into its own file because it doesn't particularly belong to any of the existing files.